### ISSUE TITLE
What is the .json config file for pre-trained sentibert?

### ISSUE THREAD

**Issue by plkmo** [WadeYin9712/SentiBERT#1]
Hi, thanks for the nice work. I tried to run Sentibert finetuning with emocontext:

CUDA_VISIBLE_DEVICES=0 python examples/run_classifier_new.py \
  --task_name emocontext \
  --do_train \
  --do_eval \
  --do_lower_case \
  --data_dir datasets/emocontext \
  --pretrain_dir results/sstphrase_pretrain/ \
  --bert_model bert-base-uncased \
  --max_seq_length 128 \
  --train_batch_size 16 \
  --learning_rate 0.001 \
  --num_train_epochs 10 \
  --output_dir results/emocontext \
  --seed 1 \
  --para sentibert

But ran into the following error:
ERROR - pytorch_pretrained_bert.modeling_new -   Model name 'results/sstphrase_pretrain/' was not found in model name list (bert-base-uncased, bert-large-uncased, bert-base-cased, bert-large-cased, bert-base-multilingual-uncased, bert-base-multilingual-cased, bert-base-chinese, bert-base-german-cased, bert-large-uncased-whole-word-masking, bert-large-cased-whole-word-masking, bert-large-uncased-whole-word-masking-finetuned-squad, bert-large-cased-whole-word-masking-finetuned-squad, bert-base-cased-finetuned-mrpc). We assumed 'results/sstphrase_pretrain/sstphrase.json' was a path or url but couldn't find any file associated to this path or url.

It looks like sstphrase.json is missing. I only managed to obtain sstphrase.bin from your download link. Can you please also provide the .json config file? Thanks.

**Comment by WadeYin9712** [comment]
Hi plkmo,

Thanks a lot for your interest in our work! The link of .json and vocab.txt are as follows:
json: https://drive.google.com/file/d/1Ra-vOaDa-fYAIqQr4w6DkK4lpolUHK6r/view?usp=sharing;
vocab.txt: https://drive.google.com/file/d/1QwnTiomdFKL6Ont_xrARjCXwk526qdHb/view?usp=sharing

Actually the json and vocab.txt can be used for all the tasks since they share the same hyperparameters of model architecture and vocabulary list. You can change the name like sst-2.json, emocontext.json, joy.json to adapt to other tasks.

Please let me know if you have other questions.

**Comment by plkmo** [comment]
Thanks! The training works now.

**Comment by Edward-Joker** [comment]
Hello，I am running this project in a windows environment. During the debugging process, I found  some problems. So I did this 👍  
In pregenerate_training_data_sstphrase.py and finetune_on_pregenerated_sstphrase.py
 span = np.load(os.path.join(args.data_dir, "sstphrase_train_span.npy"),allow_pickle=True)#add 
 allow_pickle=True，**other np.load() function also do this!!!**  Because I see the source code default=Flase.

I am running task sstphrase in pycharm，The parameters are set like this:
**Firstly！**
run sstphrase_st.py 
For SST-phrase:
--data_dir
../datasets/sstphrase/
--tree_dir
../datasets/sstphrase/
--stage
train

run sstphrase_st.py 
For SST-phrase:
--data_dir
../datasets/sstphrase/
--tree_dir
../datasets/sstphrase/
--stage
test
**Second！For sstphrase task**

Pretrain the generated epochs: pregenerate_training_data_sstphrase.py 
--pregenerated_data
/training_sstphrase
--bert_model
bert-base-uncased
--do_lower_case
--output_dir
/results/sstphrase_pretrain
--epochs
3

Generate epochs for preparation: run finetune_on_pregenerated_sstphrase.py
 --train_corpus
 ../../datasets/sstphrase/sstphrase_train_text_new.txt
 --data_dir
 ../../datasets/sstphrase/
 --bert_model
 bert-base-uncased
 --do_lower_case
 --output_dir
 /training_sstphrase
 --epochs_to_generate
 3
 --max_seq_len
 128



**Finally！**

run run_classifier_new.py
--task_name
sstphrase
--do_train
--do_eval
--do_lower_case
--data_dir
../datasets/sstphrase
--pretrain_dir
/results/sstphrase_pretrain
--bert_model
bert-base-uncased
--max_seq_length
128
--train_batch_size
8
--output_dir
/results/sstphrase

Maybe the code has other bug in the windows environment.
I think this is an interesting idea，this model incorporates contextualized representation with binary constituency parse tree to capture semantic composition. This is the first time I have seen the use of tree structure to capture semantic information.

Last question：Do you use distributed GPUs for training?

**Comment by WadeYin9712** [comment]
Hi Edward,

Thanks a lot for your interest! Actually we just implemented in Linux environment. We've not tried it on Windows. Sorry for annoying you using our code in windows environment.

We used GPU to train our model.

Thanks!

**Comment by Edward-Joker** [comment]
Thank you for your reply，I will read your project code carefully！
